L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetCalibrator.cc

```diff
@@ -0,0 +1,187 @@
+  // 4 run a dicotomic search on the pt binning to find the pt index
+  // 5 fetch the calibration factor
+  // 6 update the candidate pt by applying the calibration factor
+  // 7 store calibrated candidate in a new collection
+  for (const auto& candidate : *inputCollectionHandle) {
+    // 1
+    float pt = candidate.pt();
+    float eta = candidate.eta();
+
+    //2
+    auto etaBin = upper_bound(absEtaBinning_.begin(), absEtaBinning_.end(), fabs(eta));
+    int etaIndex = etaBin - absEtaBinning_.begin() - 1;
+
+    //3
+    const std::vector<double>& l1tPtBins = _jetCalibrationFactorsPtBins[etaIndex];
+    const std::vector<double>& l1tCalibrationFactors = _jetCalibrationFactorsBinnedInEta[etaIndex];
+
+    //4
+    auto ptBin = upper_bound(l1tPtBins.begin(), l1tPtBins.end(), pt);
+    int ptBinIndex = ptBin - l1tPtBins.begin() - 1;
+
+    //5
+    const double& l1tCalibrationFactor = l1tCalibrationFactors[ptBinIndex];
+
+    //6
+    reco::Candidate::PolarLorentzVector candidateP4(candidate.polarP4());
+    reco::CaloJet* newCandidate = candidate.clone();
+    candidateP4.SetPt(candidateP4.pt() * l1tCalibrationFactor);
+    newCandidate->setP4(candidateP4);
+
+    //7
+    calibratedCollectionPtr->emplace_back(*newCandidate);
+    // clean up
+
+#ifdef DEBUG
+    if (newCandidate->pt() < 0) {
+      std::cout << "######################" << std::endl;
+      std::cout << "PRE-CALIBRATION " << std::endl;
+      std::cout << "\t Jet properties (pt, eta, phi, pile-up): " << candidate.pt() << "\t" << candidate.eta() << "\t"
+                << candidate.phi() << "\t" << candidate.pileup() << std::endl;
+      std::cout << "CALIBRATION " << std::endl;
+      std::cout << "\t Using eta - pt - factor " << *etaBin << " - " << *ptBin << " - " << l1tCalibrationFactor
+                << std::endl;
+      std::cout << "POST-CALIBRATION " << std::endl;
+      std::cout << "\t Jet properties (pt, eta, phi, pile-up): " << newCandidate->pt() << "\t" << newCandidate->eta()
+                << "\t" << newCandidate->phi() << "\t" << newCandidate->pileup() << std::endl;
+    }
+#endif
+
+    delete newCandidate;
+  }
+
+  // finally, sort the collection by pt
+  std::sort(calibratedCollectionPtr->begin(),
+            calibratedCollectionPtr->end(),
+            [](const reco::CaloJet& jet1, const reco::CaloJet& jet2) { return jet1.pt() > jet2.pt(); });
+
+  iEvent.put(std::move(calibratedCollectionPtr), outputCollectionName_);
+}
+
+// ------------ method called once each stream before processing any runs, lumis or events  ------------
+void Phase1L1TJetCalibrator::beginStream(edm::StreamID) {}
+
+// ------------ method called once each stream after processing all runs, lumis and events  ------------
+void Phase1L1TJetCalibrator::endStream() {}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void Phase1L1TJetCalibrator::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
```

Comment:
not a good idea.  configuration parameters are all calibration constants. too many of them.  In the future this can go into a DB, but for now in the cfi file.

Reply:
Added here without default values for calibration constants: 
https://github.com/cms-l1t-offline/cmssw/blob/fc3a1a454f40efdf7c95dd97295a6a085a674092/L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetCalibrator.cc#L169-L183